modules/tensor_mechanics/src/kernels/DynamicStressDivergenceTensors.C

```diff
@@ -0,0 +1,138 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#include "DynamicStressDivergenceTensors.h"
+
+#include "Material.h"
+
+template<>
+InputParameters validParams<DynamicStressDivergenceTensors>()
+{
+  InputParameters params = validParams<Kernel>();
+  params.addClassDescription("Rayleigh damping and HHT time integration term to be used along with Stress divergence kernel in Tensor Mechanics module");
+  params.addRequiredParam<unsigned int>("component", "An integer corresponding to the direction the variable this kernel acts in. (0 for x, 1 for y, 2 for z)");
+  params.addCoupledVar("displacements", "The string of displacements suitable for the problem statement");
+  params.addCoupledVar("disp_x", "Depricated: the x displacement");
+  params.addCoupledVar("disp_y", "Depricated: the y displacement");
+  params.addCoupledVar("disp_z", "Depricated: the z displacement");
+  params.addCoupledVar("temp", "The temperature");
+  params.addParam<Real>("zeta", 0, "zeta parameter for the Rayleigh damping");
+  params.addParam<Real>("alpha", 0, "alpha parameter for HHT time integration");
+  params.addParam<std::string>("base_name", "Material property base name");
+  params.set<bool>("use_displaced_mesh") = false;
+
+  return params;
+}
+
+
+DynamicStressDivergenceTensors::DynamicStressDivergenceTensors(const InputParameters & parameters) :
+    Kernel(parameters),
+    _base_name(isParamValid("base_name") ? getParam<std::string>("base_name") + "_" : ""),
+    _stress_older(getMaterialPropertyOlderByName<RankTwoTensor>(_base_name + "stress")),
+    _stress_old(getMaterialPropertyOldByName<RankTwoTensor>(_base_name + "stress")),
+    _stress(getMaterialPropertyByName<RankTwoTensor>(_base_name + "stress")),
+    _Jacobian_mult(getMaterialPropertyByName<ElasticityTensorR4>(_base_name + "Jacobian_mult")),
+    _component(getParam<unsigned int>("component")),
+    _ndisp(coupledComponents("displacements")),
+    _disp(3),
```

Comment:
We typically use LIBMESH_DIM in place of "3" in MOOSE. @dschwen, do you have any opinion here?